src/code42cli/cmds/legal_hold.py

```diff
@@ -124,6 +138,37 @@ def show(state, matter_id, include_inactive=False, include_policy=False):
         echo("")
 
 
+@legal_hold.command()
+@click.option(
+    "--matter-id", type=str, help="Filter results by legal hold UID",
+)
+@click.option(
+    "--event-type",
+    type=click.Choice(
+        [
+            "MembershipCreated",
+            "MembershipReactivated",
+            "MembershipDeactivated",
+            "HoldCreated",
+            "HoldDeactivated",
+            "HoldReactivated",
+            "Restore",
+        ]
+    ),
+    help="Filter results by event types",
+)
+@click.option("--begin", **BEGIN_DATE_DICT)
```

Comment:
This should use the `begin_option` defined in `code42cli.options`
There will be several advantages to doing this.

Reply:
Same with `end_option`

Reply:
Sure, I avoided this option initially because the `begin` option requires a cursor initialized for state, which I didn't think was necessary. I've added a LegalHoldEvents cursor to the cursor_store.py to make it work, but not sure if that's appropriate.

Reply:
I wonder if we should implement the `--use-checkpoint` option for this command and probably also make a `send-to`. As I could see customers wanting to be able to automate sending this data into a SIEM.

Reply:
Oh!  @maddie-vargo Sorry, I did not forget about that when I made my initial comments.

I do like the idea from @timabrmsn, supporting checkingpointing would be a nice feature. We can always add it later though

Reply:
@unparalleled-js  would it be possible to use the `begin_option` but somehow designate it as optional? I left the original options in there for now. 

I left checkpoint'ing off for now.